tools/cli/internal/versioning/version.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package versioning
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+)
+
+type APIVersion struct {
+	version     string
+	versionDate time.Time
+}
+
+const (
+	dateFormat = "2006-01-02"
+)
+
+// NewAPIVersion creates a new API version.
+func NewAPIVersionFromDateString(version string) (*APIVersion, error) {
```

Comment:
from https://go.dev/doc/effective_go#package-names

if your package is `apiversion` (or version) this shoudl be 
```suggestion
func NewFromDateString(version string) (*APIVersion, error) {
```